src/main/java/org/hydev/mcpm/client/installer/InstallInteractor.java

```diff
@@ -46,20 +57,25 @@ public InstallInteractor(PluginDownloader spigotPluginDownloader, DatabaseManage
      * @param installInput: the plugin submitted by the user for installing
      * */
     @Override
-    public InstallResult installPlugin(InstallInput installInput)
+    public boolean installPlugin(InstallInput installInput,
+                                       ResultPresenter resultPresenter)
     {
         // 1. Search the name and get a list of plugins
         SearchPackagesResult searchResult = databaseManager.getSearchResult(installInput);
 
-
         if (searchResult.state() != SearchPackagesResult.State.SUCCESS) {
-            return new InstallResult(
+            InstallResult installResult = new InstallResult(
                     searchResult.state() == SearchPackagesResult.State.FAILED_TO_FETCH_DATABASE ?
                             InstallResult.Type.SEARCH_FAILED_TO_FETCH_DATABASE :
                             InstallResult.Type.SEARCH_INVALID_INPUT);
+            resultPresenter.displayResult(installInput.name() + ": " + installResult.type().reason());
+            return false;
         }
+
         if (searchResult.plugins().isEmpty()) {
-            return new InstallResult(Type.NOT_FOUND);
+            InstallResult installResult = new InstallResult(Type.NOT_FOUND);
```

Comment:
^^ If you're going to get rid of InstallResult, maybe you should delete it and see if that cleans some stuff up.